### ISSUE TITLE
Anything with ac commands or mix_rgb does not work

### ISSUE THREAD

**Issue by Reptorian1125** [dtschump/gmic#103]
After finding out that some filters don't work, the common errors that comes up in every single instants of failed filters. 

Software used -
Krita 4.2 Pre-Alpha git 402c41c
G'MIC 2.4.5

Operating System-
Windows 10 x64 Home Edition

[gmic_krita_qt]./apply/ v 0 pseudo_ecb 0,0,1,20,1,1,5,0,0,5,90,0,41,1,15,256,75,16,100,1,41,1,41,1,1,0,1,41,1,0,50,50
[gmic]-1./ Start G'MIC interpreter.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/ Import custom commands from expression 'MergeChoice : $=_mode' (1 new, total: 3351).
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode0=MergeChoice'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode1=add'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode2=alpha'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode3=and'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode4=average'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode5=blue'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode6=burn'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode7=darken'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode8=difference'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode9=divide'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode10=dodge'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode11=exclusion'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode12=freeze'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode13=grainextract'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode14=grainmerge'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode15=green'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode16=hardlight'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode17=hardmix'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode18=hue'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode19=interpolation'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode20=lighten'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode21=lightness'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode22=linearburn'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode23=linearlight'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode24=luminance'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode25=multiply'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode26=negation'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode27=or'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode28=overlay'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode29=pinlight'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode30=red'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode31=reflect'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode32=saturation'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode33=screen'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode34=shapeaverage'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode35=softburn'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode36=softdodge'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode37=softlight'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode38=stamp'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode39=subtract'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode40=value'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode41=vividlight'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode42=xor'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode43=edges'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/MergeChoice/ Set global variable '_mode44=error'.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/ac/ Apply command 'ecb_gradient_base[0] 5,0,0' on channels 'rgb' of image [0].
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/*local/*if/ac/ Apply command 'ecb_gradient_base[0] 5,90,0' on channels 'rgb' of image [0].
[gmic]-2./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/*local/*if/ Blend all images [0,1] together, using 'xor' mode and opacity 1.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/*local/ Sharpen image [0] with inverse diffusion and amplitude 15.
[gmic]-1./pseudo_ecb/*repeat/*local/(...)/blend_bomb/*if/*repeat/*local/ Input copy of image [0] at position 1 (1 image 256x256x1x4).
[gmic]-2./pseudo_ecb/*repeat/*local/(...)/*if/*repeat/*local/_blend_bomb/ Force images [0,1] to be in RGBA mode.
[gmic]-2./pseudo_ecb/*repeat/*local/(...)/*if/*repeat/*local/_blend_bomb/ Input black image at position 2 (1 image 256x256x1x4).
[gmic]-3./pseudo_ecb/*repeat/*local/(...)/*if/*repeat/*local/_blend_bomb/ Add gaussian noise to image [2], with standard deviation 255.
[gmic]-3./pseudo_ecb/*repeat/*local/(...)/*if/*repeat/*local/_blend_bomb/ Resize image [2] to 256x256x100%x100% , with nearest-neighbor interpolation, dirichlet boundary conditions and alignment (0,0,0,0).
[gmic]-3./pseudo_ecb/*repeat/*local/(...)/*if/*repeat/*local/_blend_bomb/ Normalize image [2] in range [0,255].
[gmic]-3./pseudo_ecb/*repeat/*local/(...)/*if/*repeat/*local/_blend_bomb/ Blur image [2] with standard deviation 0%, neumann boundary conditions and quasi-gaussian kernel.
[gmic]-1./pseudo_ecb/*repeat/*local/(...)/*local/_blend_bomb/*local/*if/ Fill image [0] with expression 'val = i/255; (val-0.5+(atan((val-0.5)*(75/10)^3)/pi)+0.5)*255'.
[gmic]-1./pseudo_ecb/*repeat/*local/(...)/*local/_blend_bomb/*local/*if/ Normalize image [0] in range [0,255].
[gmic]-3./pseudo_ecb/*repeat/*local/(...)/*if/*repeat/*local/_blend_bomb/ Reverse positions of images [0,1,2].
[gmic]-3./pseudo_ecb/*repeat/*local/(...)/fx_mesh_blend/*if/*repeat/*local/ Reverse positions of images [0,1,2].
[gmic]-3./pseudo_ecb/*repeat/*local/(...)/*if/*repeat/*local/_fx_mesh_blend/ Resize image [2] to 256x256x100%x100% , with nearest-neighbor interpolation, dirichlet boundary conditions and alignment (0,0,0,0).
[gmic]-3./pseudo_ecb/*repeat/*local/(...)/*repeat/*local/_fx_mesh_blend/*if/ Force images [0,1,2] to be in RGB mode.
[gmic]-3./pseudo_ecb/*repeat/*local/(...)/*if/*repeat/*local/_fx_mesh_blend/ Fill image [0] with expression 'i(#2,i(#0),256-i(#1))'.
[gmic]-3./pseudo_ecb/*repeat/*local/(...)/*repeat/*local/_fx_mesh_blend/*if/ Remove image [2] (2 images left).
[gmic]-2./pseudo_ecb/*repeat/*local/(...)/*repeat/*local/_fx_mesh_blend/*if/ Remove image [1] (1 image left).
[gmic]-1./pseudo_ecb/*repeat/*local/(...)/*local/_blend_bomb/*if/ac/ Apply command 'n 0,255' on channels 'rgba' of image [0].
[gmic]-3./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/*local/ Blend all images [0,1,2] together, using 'xor' mode and opacity 1.
[gmic]-1./pseudo_ecb/*repeat/*local/(...)/blend_bomb/*if/*repeat/*local/ Input copy of image [0] at position 1 (1 image 256x256x1x3).
[gmic]-2./pseudo_ecb/*repeat/*local/(...)/*if/*repeat/*local/_blend_bomb/ Force images [0,1] to be in RGBA mode.
[gmic]-2./pseudo_ecb/*repeat/*local/(...)/*if/*repeat/*local/_blend_bomb/ Input black image at position 2 (1 image 16x16x1x4).
[gmic]-3./pseudo_ecb/*repeat/*local/(...)/*if/*repeat/*local/_blend_bomb/ Add gaussian noise to image [2], with standard deviation 255.
[gmic]-3./pseudo_ecb/*repeat/*local/(...)/*if/*repeat/*local/_blend_bomb/ Resize image [2] to 256x256x100%x100% , with nearest-neighbor interpolation, dirichlet boundary conditions and alignment (0,0,0,0).
[gmic]-3./pseudo_ecb/*repeat/*local/(...)/*if/*repeat/*local/_blend_bomb/ Normalize image [2] in range [0,255].
[gmic]-3./pseudo_ecb/*repeat/*local/(...)/*if/*repeat/*local/_blend_bomb/ Blur image [2] with standard deviation 0%, neumann boundary conditions and quasi-gaussian kernel.
[gmic]-1./pseudo_ecb/*repeat/*local/(...)/*local/_blend_bomb/*local/*if/ Fill image [0] with expression 'val = i/255; (val-0.5+(atan((val-0.5)*(100/10)^3)/pi)+0.5)*255'.
[gmic]-1./pseudo_ecb/*repeat/*local/(...)/*local/_blend_bomb/*local/*if/ Normalize image [0] in range [0,255].
[gmic]-3./pseudo_ecb/*repeat/*local/(...)/*if/*repeat/*local/_blend_bomb/ Reverse positions of images [0,1,2].
[gmic]-3./pseudo_ecb/*repeat/*local/(...)/fx_mesh_blend/*if/*repeat/*local/ Reverse positions of images [0,1,2].
[gmic]-3./pseudo_ecb/*repeat/*local/(...)/*if/*repeat/*local/_fx_mesh_blend/ Resize image [2] to 256x256x100%x100% , with nearest-neighbor interpolation, dirichlet boundary conditions and alignment (0,0,0,0).
[gmic]-3./pseudo_ecb/*repeat/*local/(...)/*repeat/*local/_fx_mesh_blend/*if/ Force images [0,1,2] to be in RGB mode.
[gmic]-3./pseudo_ecb/*repeat/*local/(...)/*if/*repeat/*local/_fx_mesh_blend/ Fill image [0] with expression 'i(#2,i(#0),256-i(#1))'.
[gmic]-3./pseudo_ecb/*repeat/*local/(...)/*repeat/*local/_fx_mesh_blend/*if/ Remove image [2] (2 images left).
[gmic]-2./pseudo_ecb/*repeat/*local/(...)/*repeat/*local/_fx_mesh_blend/*if/ Remove image [1] (1 image left).
[gmic]-2./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/ Blend all images [0,1] together, using 'xor' mode and opacity 1.
[gmic]-1./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/*if/ Input copy of image [0] at position 1 (1 image 256x256x1x4).
[gmic]-2./pseudo_ecb/*repeat/*local/*repeat/pseudo_ecb_generate/*if/ Blend all images [0,1] together, using 'lightness' mode and opacity 1.
[gmic_krita_qt]./error/ When running command 'v 0 pseudo_ecb 0,0,1,20,1,1,5,0,0,5,90,0,41,1,15,256,75,16,100,1,41,1,41,1,1,0,1,41,1,0,50,50', this error occured:
*** Error in ./pseudo_ecb/*repeat/*local/(...)/mix_rgb/*repeat/mix_channels/*if/ *** Command 'input': Invalid argument '((0.0016174760784313726,0.001402259215(...)1960784313727,0.0037266827450980391))'.

[gmic_krita_qt]./apply/ v 0 fx_stylize 5,6,0,1,0.5,3,3,0.5,0.1,2,5,0,0.7,1,0,1,0,5,5,7,1,30,1000,2,1.85,0
[gmic]-1./ Start G'MIC interpreter.
[gmic]-0./fx_stylize/*local/*if/_fx_stylize_/ Import custom commands from expression '_b:base642img $1' (1 new, total: 3351).
[gmic]-0./fx_stylize/*local/*if/_fx_stylize_/ Import custom commands from expression '_r:run {t} rm' (1 new, total: 3352).
[gmic]-0./fx_stylize/*local/*if/_fx_stylize_/_r/run/_r/ Remove image [] (0 images left).
[gmic]-1./fx_stylize/*local/*if/_fx_stylize_/_r/ Remove image [0] (0 images left).
[gmic]-0./fx_stylize/*local/*if/_fx_stylize_/ Discard definitions of custom commands '_b,_r' (2 found, 3351 commands left).
[gmic]-1./fx_stylize/ Set local variable 'init_resolution=16'.
[gmic]-1./fx_stylize/ Set local variable 'colorspace=lab'.
[gmic]-1./fx_stylize/ Set local variable 'match_colors=s c,-3 mv[1] 3 b[-2,-1] 1% negate[-2,-1] n[-2,-1] 0,1'.
[gmic]-1./fx_stylize/*if/ Set local variable 'match_colors=s c,-3 mv[1] 3 b[-2,-1] 1% negate[-2,-(...)0,1 transfer_histogram[0] [1],256,lab'.
[gmic]-1./fx_stylize/ Set local variable 'match_colors=s c,-3 mv[1] 3 b[-2,-1] 1% negate[-2,-(...) *[-2,-1] {1*255} a[0,-2] c a[1,-1] c'.
[gmic]-1./fx_stylize/*if/ Set local variable 'ind_first=0'.
[gmic]-1./fx_stylize/*if/ Set local variable 'N=1'.
[gmic]-1./fx_stylize/*if/ Set local variable 'ind_style='.
[gmic]-1./fx_stylize/*if/_fx_stylize/*if/ Set local variable 'name=themandola'.
[gmic]-1./fx_stylize/*if/_fx_stylize/ Set local variable 'filename=C:\Users\User\AppData\Roaming\gmic\style_themandola.png'.
[gmic]-1./fx_stylize/*if/_fx_stylize/*if/ Input file 'C:\Users\User\AppData\Roaming\gmic\style_themandola.png' at position 1 (1 image 842x1024x1x3).
[gmic]-2./fx_stylize/ Set local variable 'is_window=0'.
[gmic]-2./fx_stylize/*repeat/*local/ Set local variable 'nm=Layer 1'.
[gmic]-1./fx_stylize/*repeat/*local/*if/ Resize 2D image [1] to 256x256 with ratio-outside mode and interpolation type 2.
[gmic]-3./fx_stylize/*repeat/*local/*if/ Set local variable 'wsiz=256,256'.
[gmic]-3./fx_stylize/*repeat/*local/*if/ Display image [0] in 256x256 display window [0], with auto-normalization, no fullscreen and title '[G'MIC] Window #0'.
[gmic]-3./fx_stylize/*repeat/*local/*if/ Set local variable 'is_window=1'.
[gmic]-1./fx_stylize/*repeat/*local/ Stylize image [0] with style image ..
[gmic_krita_qt]./error/ When running command 'v 0 fx_stylize 5,6,0,1,0.5,3,3,0.5,0.1,2,5,0,0.7,1,0,1,0,5,5,7,1,30,1000,2,1.85,0', this error occured:
*** Error in ./fx_stylize/*repeat/*local/(...)/mix_rgb/*repeat/mix_channels/*if/ *** Command 'input': Invalid argument '((0.0016174760784313726,0.001402259215(...)1960784313727,0.0037266827450980391))'.

The same exact error appeared in those. And in every other filters that failed.

**Comment by dtschump** [comment]
Please make a simpler use case.
The command ` pseudo_ecb` is not a part of the standard library, thus we could suspect the code of this command has a bug instead (a wrong usage of ` ac` maybe ?).
Many functions and filters of the stdlib use ` apply_channel`  and ` mix_rgb` so, at a first glance I doubt these commands are the reason of the unexpected behavior.

**Comment by Reptorian1125** [comment]
I tested again, and it's probably just a few filters. Stylize do have this issue though. 

 The ECB does work fine with G'MIC 2.4.4 from my end, and nothing has been changed, so I thought there is definitely something's up here. 

Found another filter that I see having a issue. Cascading self-glitch filter. Works fine on 2.4.4.

fx_self_glitching_cascade 5,0,0,50,50,3,3,1,55,55,0,45,45,0.75,3,0,0,0,0,1,0,256,1,0,0
*** Error in ./fx_self_glitching_cascade/*repeat/*local/(...)/mix_rgb/*repeat/mix_channels/*if/ *** Command 'input': Invalid argument '((0.0016174760784313726,0.001402259215(...)1960784313727,0.0037266827450980391))'.

They all have the same numbers on the end with mix_rgb repeat.

Interesting though it has to do with Lightness. When I set to LCH usage for QAM Filter or ECB , preview timeout, and I would predict the same error. And if I click, on apply, I get the same error. Has there been any changes? So maybe, perhaps this only involves lightness?
